model_specific/SM/decays/standard_model_decays.cpp

```diff
@@ -0,0 +1,5983 @@
+                  continue;
+
+               auto const vertexId1Val = vertexId1::evaluate(indexId1, context);
+               auto const vertexId2Val = vertexId2::evaluate(indexId2, context);
+               auto const vertexId3Val = vertexId3::evaluate(indexId3, context);
+
+               if (!vertexId1Val.isZero() && !vertexId2Val.isZero() && !vertexId3Val.isZero()) {
+                  // internal masses
+                  const double mInternal1 {context.mass<typename conj<VWp>::type>(vertexId1::template indices_of_field<2>(indexId1))};
+                  const double mInternal2 {context.mass<Hp>(vertexId2::template indices_of_field<2>(indexId2))};
+                  const double mInternal3 {context.mass<typename bar<Fe>::type>(vertexId1::template indices_of_field<0>(indexId1))};
+
+                  result += symmetryFac * colorFac * calculate_diagram_SFF_t1g5n5_VSF(
+                     result.m_decay, result.m_fermion_1, result.m_fermion_2,
+                     mInternal1, mInternal2, mInternal3, 
+                     -1.0i*vertexId1Val.left(), -1.0i*vertexId1Val.right(), 1.0i*vertexId2Val.left()
+                        , 1.0i*vertexId2Val.right(), 1.0i*vertexId3Val.value(0, 1),
+                     ren_scale);
+               }
+            }
+         }
+      }
+   }
+
+   // topology T1
+   // internal particles in the diagram: conj[VWp], VWp, Fe
+   {
+      using vertexId1 = Vertex<typename bar<Fe>::type, Fv, typename conj<VWp>::type>;
+      using vertexId2 = Vertex<typename bar<Fv>::type, Fe, VWp>;
+      using vertexId3 = Vertex<hh, typename conj<VWp>::type, VWp>;
+
+      constexpr double symmetryFac {1.000000000000000};
+
+      constexpr double colorFac {1.000000000000000};
+
+      // loops over vertices' indices
+      for (const auto& indexId1: index_range<vertexId1>()) {
+         for (const auto& indexId2: index_range<vertexId2>()) {
+            for (const auto& indexId3: index_range<vertexId3>()) {
+
+               // skip indices that don't match external indices
+               const auto externalFieldIndicesIn1 = vertexId3::template indices_of_field<0>(indexId3);
+               const auto externalFieldIndicesIn2 = vertexId1::template indices_of_field<1>(indexId1);
+               const auto externalFieldIndicesIn3 = vertexId2::template indices_of_field<0>(indexId2);
+
+               if (externalFieldIndicesIn1 != idx_1 || externalFieldIndicesIn2 != idx_2 || externalFieldIndicesIn3 != idx_3)
+                  continue;
+
+               // connect internal particles in vertices
+               if (vertexId1::template indices_of_field<2>(indexId1) != vertexId3::template indices_of_field<2>(indexId3))
+                  continue;
+
+               if (vertexId2::template indices_of_field<2>(indexId2) != vertexId3::template indices_of_field<1>(indexId3))
+                  continue;
+
+               if (vertexId1::template indices_of_field<0>(indexId1) != vertexId2::template indice
```

Comment:
From looking at this (and the following) functions, it seems to me that there is an unnecessary copy here: First there is a copy of the `model` in the argument. Then there is another copy to create `context_base`. I'd propose to pass the function argument by const reference to avoid one copy.

Reply:
This also applies to the following functions.

Reply:
Sorry, where?

Reply:
Sorry, it seems that github is not pointing to the right place. I'm referring to the function `partial_width_hh_to_VGVG(Standard_model model)`, and the functions defined below that .

Reply:
Done in e1462c5.

Reply:
Thanks!